### DISCUSSION TITLE
How to use HomeKit momentary state

### DISCUSSION THREAD

**Discussion by pnuding** [rednblkx/HomeKey-ESP32#135] [General]
I'd love to have the homekey reader behave as a native lock so I can trigger HomeKit automations based on it:
1. By default the lock appears locked
2. Upon successful Homekey authentication, the homekey lock would change state to unlocked (which I can then use to trigger automations)
3. after a few seconds the lock returns to locked, taking us back to 1.

Reading the documentation I thought I need to: 
* Enable "always lock" and "always unlock" on the HomeKit settings page
* Set a random GPIO number for the last simple gpio section and enable momentary state on the actions page
Is the above correct?

However I find I cannot seem to save the changes on the actions page. I enter the GPIO and enable momentary state, try to save and it does nothing. Next time I load the page, the GPIO is back to 255 and momentary is off.
What am I doing wrong?

**Comment by pnuding** [comment]
Looks like the only missing magic was the v0.4.1 update - now I was able to save the gpio settings and it works exactly as I thought it would 🎉

**Comment by pnuding** [comment]
Sadly this doesn't seem to work as well as I hoped, and the reason seems to be HomeKit itself. 
I have set up a HomeKit automation set based on the Homekey reader's lock status - if it unlocks, my HomeKit Yale Linus lock should also unlock. This works - sometimes. Most of the time however nothing happens. 
However I find that it works 100% reliably when I look at it in the Home app on my phone.
That makes me suspect that the automation isn't executed on the home hub (an Apple TV) but rather is being handled by my phone, which may just not notice the Homekey reader unlocking if I'm not using the Home App. 
Does anybody else have experience with this?

**Comment by Tonitom** [comment]
I’m setting everything up with only HomeKit, without MQTT. I noticed that state changes sometimes don’t sync properly in HomeKit. Is there a way to automatically reset the state back to “locked”?

**Reply by pnuding** [reply]
I suspect that's the cause behind the homekit automation failures. I also have sometimes seen notifications about the reader being "locked" hours after i've used it
